### ISSUE TITLE
Tensorflow model assets import degrading performance scores randomly

### ISSUE THREAD

**Issue by Accelerator-One** [GoogleChrome/lighthouse#11128]
Hello lighthouse team,
   I'm trying to load a tjfs model locally for my web app ( in the end ) for a later point of time. All of my page metrics are fine except the fact that either lighthouse is giving me poor performance score or either giving me score of 90+ with note : " page loaded too slowly to finish within the time limit ". Is there a way I can avoid my model import to affect my performance scores since I want to start loading my model as soon as poosible after initial load.

Link : https://toxicity-detector.web.app/
( Checked with lighthouse chrome extension )

With regards,
Aditya

**Comment by patrickhulce** [comment]
Thanks for filing! I'm not sure I understand the question though @Accelerator-One . Are you asking how you can hide requests from Lighthouse so they don't affect the performance score?

**Comment by Accelerator-One** [comment]
I don't intend to hide anything in particular @patrickhulce , my apologies for an unclear question, please let me rephrase it correctly :

1. Is there any way I can let lighthouse know that I'm loading XYZ resource and it's going to be bulky due to the nature of the application itself, if possible to not consider it during evaluation.

2. Recently I observed few changes in evaluation criteria of lighthouse E.g. Addition of maskable icons as a requirement of PWA so I was hoping to get insights to not end up degrading my app evaluations by lighthouse in the future.

3. I felt lighthouse scores are sometimes a bit dependent on user's connection speed about which we developers can't do much (except trying to serve compressed data) and these scores are variable even on web.dev checker which leaves us quite ambiguous about quality of our app. I just felt the concern to be raised for consideration within the lighthouse team.

I am only trying to improve my lighthouse score as 80+ criteria described by play store for PWAs to be published as TWAs (I published it before, it was varying around 80 back then by lighthouse extension), now I am quite ambiguous and concerned due to such varying scores. I just want to restructure my app so that I can keep them on the safe side on play store.

**Comment by patrickhulce** [comment]
Thanks for the clarification.

> Is there any way I can let lighthouse know that I'm loading XYZ resource and it's going to be bulky due to the nature of the application itself, if possible to not consider it during evaluation.

There is not a way to mark a request for Lighthouse to not consider it. However, the only way a non-image network request will really impact the performance metrics is if the app ends up running script upon its completion that results in a long task or large layout. Since the tensorflow model is not this case, I would suggest deferring loading this until after the app has settled (all work is complete and `load` event has already fired).

> Addition of maskable icons as a requirement of PWA so I was hoping to get insights to not end up degrading my app evaluations by lighthouse in the future.

I'm not sure I understand what the request is here. I'm assuming how to stay up to date with what Lighthouse is doing? We publish preview releases before every major version to give early adopters a chance to see what's coming next. If you're wanting to stay ahead of the curve, perhaps follow those to receive advanced warning of new audits.

> and these scores are variable even on web.dev checker which leaves us quite ambiguous about quality of our app. I just felt the concern to be raised for consideration within the lighthouse team.

Thanks for sharing. We've [documented this concern as best we can here](https://github.com/GoogleChrome/lighthouse/blob/master/docs/variability.md). The advice in there about running multiple times and limiting non-deterministic behavior applies to all environments, web.dev and PSI included.

**Comment by Accelerator-One** [comment]
Thanks for addressing the query. Henceforth, I'll be closing the issue here.
